src/VisualStudio/Core/Def/Implementation/Workspace/SourceGeneratedFileManager.cs

```diff
@@ -302,6 +283,14 @@ public OpenSourceGeneratedFile(SourceGeneratedFileManager fileManager, ITextBuff
                     _cancellationTokenSource.Token);
             }
 
+            private void DisconnectFromWorkspaceIfOpen()
+            {
+                if (_workspace.IsDocumentOpen(_documentIdentity.DocumentId))
```

Comment:
any threading assertion here?  this feels racey.

Reply:
Assertion added, good catch.